src/library.js

```diff
@@ -5239,6 +5239,34 @@ LibraryManager.library = {
     return thisProgram || './this.program';
 #endif
   },
+
+  $listenOnce: function(object, event, func) {
+#if MIN_CHROME_VERSION < 55 || MIN_EDGE_VERSION < 18 || MIN_FIREFOX_VERSION < 50 || MIN_IE_VERSION != TARGET_NOT_SUPPORTED // https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener
+    object.addEventListener(event, function handler() {
+      func();
+      object.removeEventListener(event, handler);
+    });
+#else
+    object.addEventListener(event, func, { 'once': true });
+#endif
+  },
+
+  $autoResumeAudioContext__deps: ['$listenOnce'],
+  $autoResumeAudioContext: function(ctx) {
+    var canvas = document.getElementById('canvas');
```

Comment:
Why do we listen on input events on a canvas object? Audio is not related to visual presentation, there may be applications that use OpenAL, that do not use any WebGL or 2D rendering or similar? 

Also, `document.getElementById('canvas');` captures the first found canvas element. That might not be the one that is appropriate, or it might not exist by the time that the page loads up audio subsystem, e.g. if it dynamically creates a canvas on video init (later).

Does this scheme work if instead one listens on the window object? If not, I think we should have the function take in the dom element object to listen on, so that callers can at least configure the element they need to listen on. Maybe OpenAL and SDL do not have anything better to choose than document.getElementById('canvas');, but other users of this function could (this is after all a generic function that is proposed to be placed in core library.js for all users)

Reply:
> Why do we listen on input events on a canvas object? 

Just to work properly when say a game has the user click on the canvas, and where the canvas doesn't let a click propagate out. This won't always work, as you said there might not be a canvas or multiple ones, but it's a common case that is nice to work out of the box.

> I think we should have the function take in the dom element object to listen on, so that callers can at least configure the element they need to listen on. 

Good idea, I made it get a list of elements to listen on (but default to document + canvas).